blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/redirect/RedirectFilter.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.blueocean.service.embedded.redirect;
+
+import hudson.Extension;
+import hudson.init.Initializer;
+import hudson.model.User;
+import hudson.util.PluginServletFilter;
+import jenkins.model.GlobalConfiguration;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Extension
+public class RedirectFilter implements Filter {
+    @Initializer(fatal = false)
+    public static void init() throws ServletException {
+        PluginServletFilter.addFilter(new RedirectFilter());
+    }
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        //noop
+    }
+
+    @Override
+    public void destroy() {
+        //noop
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest request1 = (HttpServletRequest) request;
+        if (request instanceof HttpServletRequest && "/".equals(request1.getPathInfo()) && request.getParameter("noDefaultRedirect") == null) {
```

Comment:
Too late, cast already done the line above. The `instanceof` should be done before.

Reply:
YIKES... good catch. That would be ugly blowing up.

Reply:
Is it possible to use a filter chain pattern to do this URL matching (https://docs.spring.io/spring-security/site/docs/3.0.x/reference/security-filter-chain.html - `    <url-pattern>/*</url-pattern>`? That way, you don't have to have the ugly path matching here and it would alleviate @jtnord 's concerns that the filter gets executed on every page.

Reply:
--> Not knowing how Jenkins/plugins interact with HttpServlets etc, this SO question _may_ help if we don't have access to the web.xml file: https://stackoverflow.com/questions/35434978/how-to-specify-url-pattern-for-servlet-filters-in-webapplicationinitializer

Reply:
The way that [PluginServletFilter](http://javadoc.jenkins.io/index.html?hudson/util/PluginServletFilter.html) is implemented and stapler I do not believe it is possible to add `Filters` to specific paths.

Reply:
Boooo - that's a shame